### PR TITLE
Swift support

### PR DESCRIPTION
**Pull request by rotemmiz**
Currently, users who use swift in their projects may not be able to use Detox if their swift version is different than the one shipped in Detox.framework.
From Detox@5.7.0 we will start compiling Detox.framework on the user's machine, to better support different Swift versions. 

Relates to #152

### CODE REVIEW

File: detox/local-cli/detox-init.js [hunk 1]
```diff
@@ -1,36 +1,37 @@
...
-const mochaTemplates = require('./templates/mocha.js')
+const mochaTemplates = require('./templates/mocha.js');
 
 program
   .option('-r, --runner [runner]', 'Test runner (currently supports mocha)', 'mocha')
   .parse(process.argv);
 
 function createFile(dir, content) {
-    try {
+  try {
```
Comment (DanielMSchmidt) [inline detox/local-cli/detox-init.js]: We should really add in prettier, these changes just obfuscate the diff.

File: detox/local-cli/detox-clean-framework-cache.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+const path = require('path');
+const os = require('os');
+
+if (os.platform() === 'darwin') {
+  const frameworkPath = path.join(os.homedir(), '/Library/Detox');
+  console.log(`Removing framework binaries from ${frameworkPath}`);
```
Comment (DanielMSchmidt) [inline detox/local-cli/detox-clean-framework-cache.js]: Why not use our default log library?

File: detox/scripts/build.sh [hunk 3]
```diff
@@ -1,14 +1,18 @@
-#!/bin/bash
+#!/bin/bash -e
```
Comment (DanielMSchmidt) [inline detox/scripts/build.sh]: 👍

**Comment by chadlwilson** [comment]
@rotemmiz Is this fix likely to address app crashes on start like the below; which I started getting after an XCode 9 (and implicit Swift 3.2) upgrade? Or should I open a ticket? Just trying to avoid noise for you all if possible.

```
Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    DYLD, [0x4] Symbol missing

Application Specific Information:
dyld: launch, loading dependent libraries
DYLD_FALLBACK_LIBRARY_PATH=/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 10.3.simruntime/Contents/Resources/RuntimeRoot/usr/lib
DYLD_ROOT_PATH=/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 10.3.simruntime/Contents/Resources/RuntimeRoot
DYLD_FALLBACK_FRAMEWORK_PATH=/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 10.3.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks
DYLD_INSERT_LIBRARIES=/Users/SNIP/SNIP/node_modules/detox/Detox.framework/Detox
DYLD_SHARED_REGION=avoid

Dyld Error Message:
  Symbol not found: __TMp10Foundation13CustomNSError
  Referenced from: /Users/USER/*/Detox.framework/Frameworks/libswiftCoreLocation.dylib
  Expected in: /Users/USER/Library/Developer/CoreSimulator/Devices/8D00526A-B1E0-43FF-9A77-95AB906664D7/data/Containers/Bundle/Application/752CDBE6-5D02-420A-BAC9-87232F20598A/workbench.app/Frameworks/libswiftFoundation.dylib
 in /Users/USER/*/Detox.framework/Frameworks/libswiftCoreLocation.dylib
```

**Pull request merged**